overlord/snapstate/snapstate.go

```diff
@@ -1394,6 +1394,13 @@ func Remove(st *state.State, name string, revision snap.Revision) (*state.TaskSe
 		chain = ts
 	}
 
+	if removeAll {
+		// disconnect interfaces and run disconnect hooks
+		disconnect := st.NewTask("disconnect-interfaces", fmt.Sprintf(i18n.G("Disconnect interfaces of snap %q"), snapsup.Name()))
```

Comment:
this generate only hooks right?  because actual disconnecting is done an masse?

it also seems too early, are the services running when we do connect on install or not yet? if they are not running this also should happen with services stopped?

Reply:
Good point Yes, it's only hooks, disconnecting is done via discard-conns. I've changed the order now to run after stop-services.